File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Circle.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Circle implements Shape {
    public double radius;

    public Circle(double radius) {
        this.radius = radius;
    }

    public double getArea() {
        return Math.PI * radius * radius;
    }
    public int getShape()
    {
        return 2;
    }

    public double[] getSize()
    {
        return new double[]{radius};
    }

}

File: typescript/src/shapes/circle.ts
import { Shape,ShapeType } from "./shape.js";


function Circle(radius: number): Shape {

    return {
        
        computeArea():  number {
            return Math.PI * radius * radius
        },
        getShape(): ShapeType {
            return 'circle';
          },
        getSize(): number[] {
            return [radius];
          },
        };
}

export { Circle}
